test/functional/fixtures/runner/browser-dialogs/before-unload-dialog-test.js

```diff
@@ -0,0 +1,34 @@
+var expect = require('chai').expect;
+
+describe('Native dialogs handling', function () {
+    it('Should succeed when expected before unload dialog appeared', function () {
+        return runTests('before-unload-dialog.test.js', 'Before unload dialog - should succeed')
+            .then(function (err) {
+                expect(err).eql('');
+            });
+    });
+
+    it('Should fail when expected before unload dialog did not appear', function () {
```

Comment:
Replace the text with "fails if the expected beforeUnload dialog does not appear"